File: backend/app/schemas/table.py
from datetime import datetime

from pydantic import BaseModel


class ColumnResponse(BaseModel):
    id: str
    name: str
    data_type: str
    length: int | None = None
    nullable: bool = True
    default_value: str | None = None
    is_primary_key: bool = False
    ordinal_position: int = 0
    comment: str | None = None

    model_config = {"from_attributes": True}


class IndexResponse(BaseModel):
    id: str
    name: str
    unique: bool = False
    columns: list[str]

    model_config = {"from_attributes": True}


class ForeignKeyResponse(BaseModel):
    id: str
    columns: list[str]
    ref_table_name: str
    ref_columns: list[str]
    constraint_name: str | None = None

    model_config = {"from_attributes": True}


class TableResponse(BaseModel):
    id: str
    name: str
    schema_name: str = ""
    comment: str | None = None
    column_count: int = 0
    created_at: datetime

    model_config = {"from_attributes": True}


class TableDetailResponse(BaseModel):
    id: str
    name: str
    schema_name: str = ""
    comment: str | None = None
    columns: list[ColumnResponse] = []
    indexes: list[IndexResponse] = []
    foreign_keys: list[ForeignKeyResponse] = []
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/table.ts
export interface TableSummary {
  id: string
  name: string
  schema_name?: string
  comment?: string
  column_count: number
  created_at: string
}

export interface Column {
  id: string
  name: string
  data_type: string
  length?: number
  nullable: boolean
  default_value?: string
  is_primary_key: boolean
  ordinal_position: number
  comment?: string
}

export interface Index {
  id: string
  name: string
  unique: boolean
  columns: string[]
}

export interface ForeignKey {
  id: string
  columns: string[]
  ref_table_name: string
  ref_columns: string[]
  constraint_name?: string
}

export interface TableDetail extends TableSummary {
  columns: Column[]
  indexes: Index[]
  foreign_keys: ForeignKey[]
}
